ebr_trackerbot/bot.py

```diff
@@ -0,0 +1,144 @@
+"""
+Slack Bot implementation
+"""
+
+import asyncio
+import os
+import re
+import sys
+import logging
+
+sys.path.append(os.path.dirname(os.path.realpath(__file__)))
+
+from state import STATE
+from checker import check_tests
+import slack
+
+
+def register_command(command, description, regexp_match, callback):
+    """
+    Register slack bot command
+    """
+    if not hasattr(STATE, "bot_command"):
+        STATE.bot_command = []
+    STATE.bot_command.append(
+        {"command": command, "description": description, "regexp_match": regexp_match, "callback": callback}
+    )
+
+
+def register_storage(name, save, load_for_user, load_all_tracked_tests, delete_for_user, clean_expired_tracks):
+    """
+    Register slack bot storage
+    """
+    if not hasattr(STATE, "bot_storage"):
+        STATE.bot_storage = []
+    STATE.bot_storage = [x for x in STATE.bot_storage if x["name"] != name]
+    STATE.bot_storage.append(
+        {
+            "name": name,
+            "save": save,
+            "load_for_user": load_for_user,
+            "load_all_tracked_tests": load_all_tracked_tests,
+            "delete_for_user": delete_for_user,
+            "clean_expired_tracks": clean_expired_tracks,
+        }
+    )
+
+
+def get_storage_name():
+    """
+    Retrieve storage name
+    """
+    return os.environ["BACKEND"] if "BACKEND" in os.environ else "memory"
+
+
+def get_storage():
+    """
+    Returns slack bot storage
+    """
+    if not hasattr(STATE, "bot_storage"):
+        STATE.bot_storage = []
+    return list(filter(lambda x: x["name"] == get_storage_name(), STATE.bot_storage))[0]
+
+
+def say_hello(**payload):
```

Comment:
Can we make this a better name, that is more reflective of what it does?

Reply:
ok